sdk/tools/mkshelllink/mkshelllink.c

```diff
@@ -226,14 +276,27 @@ int main(int argc, const char *argv[])
         ID_LIST_DRIVE IdListDrive;
         unsigned cbListSize = sizeof(IdListGuid) + sizeof(uint16_t), cchName;
         const char *pszName = pszTarget;
+        int index = 1, specialindex = -1;
+        const struct SPECIALFOLDER *special = get_special_folder(pszTarget);
 
         // ID list
         // It seems explorer does not accept links without id list. List is relative to desktop.
 
-        pszName = pszTarget;
+        if (special)
+        {
+            Header.Flags &= ~LINK_RELATIVE_PATH;
+            CsidlBlock.cbSize = sizeof(CsidlBlock);
+            CsidlBlock.dwSignature = EXP_SPECIAL_FOLDER_SIG;
+            CsidlBlock.idSpecialFolder = special->csidl;
+            specialindex = 3; // Skip GUID, drive and fake windows/reactos folder
+            pszTarget += sizeof("shell:") - 1 + strlen(special->name) - 10;
+            memcpy((char*)pszTarget, "x:\\reactos\\", 11); // cast away const (we are really writing to argv)
```

Comment:
That is a really bad idea...

I'd much prefer it if you would allocate a new buffer for this...